users/greatwizard/underglow.c

```diff
@@ -0,0 +1,138 @@
+/* Copyright 2020 Guillaume Gérard
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "underglow.h"
+
+void keyboard_post_init_rgb(void) {
+    user_config_t user_config;
+    user_config.raw = eeconfig_read_user();
+    if (user_config.rgb_layer_change) {
+        rgblight_enable_noeeprom();
+        rgblight_setrgb(RGBLIGHT_COLOR_LAYER_1);
```

Comment:
Instead of calling this, it would be better to call `layer_state_set_rgb(layer_state)`.  This way, the layer state function sets the default color.